lib/Ceilometer/Types.hs

```diff
@@ -1 +1,111 @@
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE RankNTypes        #-}
+
 module Ceilometer.Types where
+
+--import           Control.Isomorphism.Partial.Iso
+import           Data.Bimap                      (Bimap)
+import qualified Data.Bimap                      as B
+import           Data.HashMap.Strict             (HashMap)
+import qualified Data.HashMap.Strict             as H
+import           Data.Text                       (Text)
+import qualified Data.Text                       as T
+import           Data.Word
+
+import           Marquise.Client
+import           Vaultaire.Types
+
+import           Ceilometer.Types.Base
+import           Ceilometer.Types.IP
+import           Ceilometer.Types.Volume
+
+import Control.Lens.Prism
+
+type Env = (FlavorMap, SourceDict)
+
+lookupMetricName = lookupSource "metric_name"
+lookupEvent      = lookupSource "_event"
+
+type CSimple = SimplePoint
+
+-- | Views the raw @SimplePoint@ as-is.
+cSimple :: Env -> Prism' SimplePoint CSimple
```

Comment:
The idea was to define prisms to view raw ``SimplePoint`` as decoded points, and also going backwards (for some publisher thing) and compose those prisms in a big function below.

I've scraped this idea and will remove the prism code, instead I think we should use plain old Control.Lens.Iso as decoding these points are not like parsing JSON blobs of unknown length and shape.